spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/SpannerRepository.java

```diff
@@ -16,14 +16,16 @@
 
 package org.springframework.cloud.gcp.data.spanner.repository;
 
+import com.google.cloud.spanner.Key;
+
 import org.springframework.cloud.gcp.data.spanner.core.SpannerOperations;
 import org.springframework.data.repository.PagingAndSortingRepository;
 
 /**
  * @author Ray Tsang
  * @author Chengyuan Zhao
  */
-public interface SpannerRepository extends PagingAndSortingRepository {
+public interface SpannerRepository<T> extends PagingAndSortingRepository<T, Key> {
```

Comment:
Limiting to `Key` as Id takes you the possibility to introduce user-defined primary key classes.

Reply:
I thought about this as well, but ultimately I don't think this takes away any generality or flexibility from the user. In addition to the points I made in the previous long comment reply, I want to add:
Even when we had the ID type as a type parameter, we ultimately relied on that ID type being acceptable into the Spanner Key type. In other words, we would do `Key.of(userGivenIdType)` and hope that it is a valid type for Key. By locking down that ID type to Key we are just being more upfront with this requirement, but not actually being more restrictive.